pre_commit/commands/hook_impl.py

```diff
@@ -202,8 +206,20 @@ def _run_ns(
     _check_args_length(hook_type, args)
     if hook_type == 'pre-push':
         return _pre_push_ns(color, args, stdin)
-    elif hook_type in {'commit-msg', 'prepare-commit-msg'}:
+    elif hook_type in 'commit-msg' or hook_type == 'prepare-commit-msg' and \
+            len(args) == 1:
         return _ns(hook_type, color, commit_msg_filename=args[0])
+    elif hook_type == 'prepare-commit-msg' and len(args) > 1:
+        if (len(args) == 2):
+            return _ns(
+                hook_type, color, commit_msg_filename=args[0],
+                commit_msg=args[1],
+            )
+        else:
+            return _ns(
+                hook_type, color, commit_msg_filename=args[0],
+                commit_msg=args[1], commit_sha=args[2],
+            )
```

Comment:
backslashes are gross and parens are unneeded -- I would write this as 4 disparate conditions:

```
elif commit-msg: ...
elif prepare-commit-msg and len(...) == 1: ...
elif prepare-commit-msg and len(...) == 2: ...
elif prepare-commit-msg and len(...) == 3: ...
```